### ISSUE TITLE
Creating KMS ReplicaKey "failed" but key was actually created

### ISSUE THREAD

**Issue by cynicaljoy** [pulumi/pulumi-aws-native#184]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details

Attempting to create a ReplicaKey for a KMS Multi-Region CMK. When running `pulumi up` my Stack update threw errors:

```
 aws-native:kms:ReplicaKey (us-west-2-replica):
    error: operation CREATE failed with "InvalidRequest": 1 validation error detected: Value null at 'keyId' failed to satisfy constraint: Member must not be null (Service: Kms, Status Code: 400, Request ID: c81e7588-25cb-40b6-a80b-b71a3b8ab6e3, Extended Request ID: null)
 
  pulumi:pulumi:Stack (<removed>):
    error: update failed
```

However, the Replica Keys were actually created. But now I can't update my stack due to:

```
  aws-native:kms:ReplicaKey (us-west-2-replica):
    error: operation CREATE failed with "AlreadyExists": You can create only one replica of each primary key in each AWS Region. (Service: Kms, Status Code: 400, Request ID: 0147eb0e-065f-491d-8a5d-665b03e01c92, Extended Request ID: null)
```

I tried to import the key to get around the issue, but that failed too: 

```
$ pulumi import aws-native:kms:ReplicaKey us-west-2-replica <removed>
Previewing import (<removed>)

View Live: <removed>

     Type                          Name                                Plan       Info
     pulumi:pulumi:Stack           <removed>             1 error; 1 message
 =   └─ aws-native:kms:ReplicaKey  us-west-2-replica                   import     1 error
 
Diagnostics:
  aws-native:kms:ReplicaKey (us-west-2-replica):
    error: Preview failed: failed to fetch provider schema: rpc error: code = Internal desc = grpc: error while marshaling: string field contains invalid UTF-8
 
  pulumi:pulumi:Stack (<removed>):
    ERROR: 2021/10/20 12:28:52 [core] grpc: server failed to encode response:  rpc error: code = Internal desc = grpc: error while marshaling: string field contains invalid UTF-8
 
    error: preview failed
```

### Steps to reproduce

1. Create Multi-Region Key
2. Add ReplicaKey

Expected: No errors, keys created. 🙌 
Actual: Errors, but key is created, but can't associate to Pulumi Stack so I'm blocked

**Comment by mikhailshilkov** [comment]
@cynicaljoy Apologies for the trouble and a slow response here!

Could you please share your code for your KMS setup so that I could repro the problem?

**Comment by cynicaljoy** [comment]
I'm unable to share publicly, sorry.

**Comment by mikhailshilkov** [comment]
Can you simplify and obfuscate?

**Comment by mikhailshilkov** [comment]
Okay I think I can get a repro from another source, I'll try to see what's going on

**Comment by mikhailshilkov** [comment]
The issue of an orphaned resource was fixed in 0.4.0: from now one, if a resource deployment partially fails, we will still persist it in the state. On the next update, the preview will show an update for it and `up` will try to update the resource to make sure it's entirely correct.

In my test, this unblocks the scenario above (the second update succeeds and the key replica is now managed by Pulumi).

> Value null at 'keyId' failed to satisfy constraint: Member must not be null

This error has been reported to AWS and the service team is working on a fix. We expect that no further action is required from the Pulumi provider, so I'll go ahead and resolve this issue. Please report back if any questions.
